test/SILGen/moveonly.swift

```diff
@@ -0,0 +1,177 @@
+// RUN: %target-swift-emit-silgen -enable-experimental-move-only %s | %FileCheck %s
+
+//////////////////
+// Declarations //
+//////////////////
+
+@_moveOnly
+public class Klass {
+    var intField: Int
+    var klsField: Klass?
+
+    // CHECK-LABEL: sil hidden [ossa] @$s8moveonly5KlassCACycfc : $@convention(method) (@owned Klass) -> @owned Klass {
+    // CHECK: bb0([[ARG:%.*]] : @owned $Klass):
+    // CHECK:   [[MARK_UNINIT:%.*]] = mark_uninitialized [rootself] [[ARG]]
+    // CHECK:   [[MARK_NO_IMP_COPY:%.*]] = mark_must_check [no_implicit_copy] [[MARK_UNINIT]]
+    // CHECK: } // end sil function '$s8moveonly5KlassCACycfc'
+    init() {
+        klsField = Klass()
+        intField = 5
+    }
+}
+
+public func nonConsumingUseKlass(_ k: Klass) {}
+
+@_moveOnly
+public struct NonTrivialStruct {
+    var k = Klass()
+}
+
+public func nonConsumingUseNonTrivialStruct(_ s: NonTrivialStruct) {}
+
+@_moveOnly
+public enum NonTrivialEnum {
+    case first
+    case second(Klass)
+    case third(NonTrivialStruct)
+}
+
+public func nonConsumingUseNonTrivialEnum(_ e : NonTrivialEnum) {}
+
+///////////
+// Tests //
+///////////
+
+//===---
+// Function Arguments
+//
+
+// CHECK-LABEL: sil [ossa] @$s8moveonly8useKlassyyAA0C0CF : $@convention(thin) (@guaranteed Klass) -> () {
+// CHECK: bb0([[ARG:%.*]] : @guaranteed $Klass):
+// CHECK:   [[OWNED_ARG:%.*]] = copy_value [[ARG]]
+// CHECK:   [[MARKED_OWNED_ARG:%.*]] = mark_must_check [no_copy] [[OWNED_ARG]]
+// CHECK: } // end sil function '$s8moveonly8useKlassyyAA0C0CF'
+public func useKlass(_ k: Klass) {
+    nonConsumingUseKlass(k)
+    let k2 = k
+    nonConsumingUseKlass(k)
+    let _ = k2
+}
+
+// CHECK-LABEL: sil [ossa] @$s8moveonly15useKlassConsumeyyAA0C0CnF : $@convention(thin) (@owned Klass) -> () {
+// CHECK: bb0([[ARG:%.*]] : @owned $Klass):
+// TODO: Is this move_value [lexical] necessary?
```

Comment:
>  TODO: Is this move_value [lexical] necessary?

The move_value is not necessary. Function arguments have implicit "lexical" scope

Reply:
I did this b/c I was worried that we had baked in this notion that owned arguments needed an implicit lexical scope. This can be seen by how we for other cases insert a begin_borrow [lexical] on the owned argument.

Reply:
I am fine fixing this in a follow on commit.